rotkehlchen/db/snapshots.py

```diff
@@ -297,13 +314,34 @@ def _import_snapshot(
                     usd_value=str(FVal(entry['usd_value'])),
                 ),
             )
+
         try:
             self.db.add_multiple_balances(processed_balances_list)
             self.db.add_multiple_location_data(processed_location_data_list)
         except InputError as err:
             return False, str(err)
         return True, ''
 
+    def update(
+        self,
+        timestamp: Timestamp,
+        balances_snapshot: List[DBAssetBalance],
+        location_data_snapshot: List[LocationData],
+    ) -> Tuple[bool, str]:
+        """Updates a snapshot of the database at a given timestamp."""
+        # delete the existing snapshot of that timestamp
+        is_success, message = self.delete(timestamp)
```

Comment:
I would move this logic outside the database since it performs complex operations (not only read/write to(from) the db). Sounds like a good logic for the `rest.py`.

Reply:
The only thing it does is read & write to the db. Unless I'm missing something here